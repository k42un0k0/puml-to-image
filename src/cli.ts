#!/usr/bin/env node
import path from "path"
import * as yargs from 'yargs'
import { pumlToImage } from "./pumlToImage"

const inputDirName = "doc";
const outputDirName = 'directory_contains_actual_images';
const jarFilePath = path.resolve(__dirname, './bin/plantuml.jar')

const argv = yargs
    .scriptName("puml-to-image")
    .usage('$0 [args]', "PlantUMLファイルをpng画像へ変換します")
    .option("inputDirName", {
        alias: "i",
        description: "画像化する対象のディレクトリ名",
        default: inputDirName
    })
    .option("outputDirName", {
        alias: "o",
        description: "出力するディレクトリ名",
        default: outputDirName
    })
    .option("jarFilePath", {
        alias: "j",
        description: "PlantUMLを画像化するjarファイルのパス",
        default: jarFilePath
    })
    .help()
    .argv


pumlToImage(argv.inputDirName, argv.outputDirName, argv.jarFilePath);