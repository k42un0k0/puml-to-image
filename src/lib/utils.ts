import fs from "fs";
export const IS_WINDOWS = process.platform === "win32";

export const readdirRecursively = (dir: string) => {
  const dirents = fs.readdirSync(dir, { withFileTypes: true });
  let dirs: string[] = [];
  let files: string[] = [];
  for (const dirent of dirents) {
    if (dirent.isDirectory()) dirs = dirs.concat(`${dir}/${dirent.name}`);
    if (dirent.isFile()) files = files.concat(`${dir}/${dirent.name}`);
  }
  for (const d of dirs) {
    files = files.concat(readdirRecursively(d));
  }
  return files;
};
