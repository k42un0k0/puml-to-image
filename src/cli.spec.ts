import { mocked } from "ts-jest/utils";
import { pumlToImage } from "./pumlToImage"
import path from "path"

jest.mock("./pumlToImage")
describe("cli.js", () => {
    test("pass options to pumlToImage", () => {
        jest.isolateModules(() => {
            process.argv = ["node", 'cli.js', '-i', 'input', '-o', 'output', '-j', 'jarfile']
            require("./cli");
            expect(pumlToImage).toBeCalledTimes(1)
            expect(pumlToImage).toHaveBeenCalledWith(
                "input", "output", "jarfile"
            )
        })
    })
    test("pass long options to pumlToImage", () => {
        jest.isolateModules(() => {
            process.argv = ["node", 'cli.js', '--inputDirName', 'input2', '--outputDirName', 'output2', '--jarFilePath', 'jarfile2']
            require("./cli");
            expect(pumlToImage).toBeCalledTimes(1)
            expect(pumlToImage).toHaveBeenCalledWith(
                "input2", "output2", "jarfile2"
            )
        })
    })
    test("pass default values to pumlToImage", () => {
        jest.isolateModules(() => {
            process.argv = ["node", 'cli.js']
            require("./cli");
            expect(pumlToImage).toBeCalledTimes(1)
            expect(pumlToImage).toHaveBeenCalledWith(
                "doc", "directory_contains_actual_images", path.resolve(__dirname, "./bin/plantuml.jar")
            )
        })
    })
})