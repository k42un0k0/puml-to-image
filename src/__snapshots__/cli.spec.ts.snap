// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli.js snapshot help 1`] = `
Object {
  "stderr": "",
  "stdout": "$ C:\\\\Users\\\\k42un\\\\Documents\\\\workspace\\\\puml-to-image\\\\node_modules\\\\.bin\\\\ts-node src/cli.ts --help
puml-to-image [args]

PlantUMLファイルをpng画像へ変換します

オプション:
      --version        バージョンを表示                                   [真偽]
  -i, --inputDirName   画像化する対象のディレクトリ名        [デフォルト: \\"doc\\"]
  -o, --outputDirName  出力するディレクトリ名
                                [デフォルト: \\"directory_contains_actual_images\\"]
  -j, --jarFilePath    PlantUMLを画像化するjarファイルのパス        [デフォルト:
        \\"C:\\\\Users\\\\k42un\\\\Documents\\\\workspace\\\\puml-to-image\\\\src\\\\bin\\\\plantuml.jar\\"]
      --help           ヘルプを表示                                       [真偽]
",
}
`;
